DEHPCommon/CommonUserInterface/ViewModels/Common/LoginViewModel.cs

```diff
@@ -0,0 +1,432 @@
+                        this.dal = new WspDal();
+                        break;
+                }
+
+                this.ServerSession = new Session(this.dal, credentials);
+
+                await this.ServerSession.Open();
+
+                this.LoginSuccessfully = true;
+
+                var siteDirectory = this.ServerSession.RetrieveSiteDirectory();
+                this.BindEngineeringModels(siteDirectory);
+                this.BindSiteReferenceDataLibraries(siteDirectory);
+            }
+            catch (Exception ex)
+            {
+                this.LogMessage(ex.Message);
+
+                this.LoginFailed = true;
+            }
+        }
+
+        /// <summary>
+        /// Bind engineering models to the reactive list
+        /// </summary>
+        /// <param name="siteDirectory">The <see cref="SiteDirectory"/> top container</param>
+        private void BindEngineeringModels(SiteDirectory siteDirectory)
+        {
+            this.EngineeringModels.Clear();
+
+            foreach (var modelSetup in siteDirectory.Model.OrderBy(m => m.Name))
+            {
+                this.EngineeringModels.Add(new EngineeringModelRowViewModel(modelSetup));
+            }
+
+            this.SelectAllModels = true;
+        }
+
+        /// <summary>
+        /// Bind site reference data libraries to the reactive list
+        /// </summary>
+        /// <param name="siteDirectory">The <see cref="SiteDirectory"/> top container</param>
+        private void BindSiteReferenceDataLibraries(SiteDirectory siteDirectory)
+        {
+            this.SiteReferenceDataLibraries.Clear();
+
+            foreach (var rdl in siteDirectory.SiteReferenceDataLibrary.OrderBy(m => m.Name))
+            {
+                this.SiteReferenceDataLibraries.Add(new SiteReferenceDataLibraryRowViewModel(rdl));
+            }
+        }
+
+        /// <summary>
+        /// Executes loading of Annex-C-3 file
+        /// </summary>
+        private void ExecuteLoadSourceFile()
+        {
+            OpenFileDialog openFileDialog = new OpenFileDialog()
+            {
+                InitialDirectory = $"{AppDomain.CurrentDomain.BaseDirectory}Import\\",
+                Filter = "Zip files (*.zip)|*.zip"
+            };
+
+            var dialogResult = openFileDialog.ShowDialog();
+
+            if (dialogResult.HasValue && dialogResult.Value && openFileDialog.FileNames.Length == 1)
+            {
+                this.Uri = openFileDialog.FileNames[0];
+            }
+        }
+
+        /// <summary>
+        /// Select model for the migration procedure
+        /// </summary>
+        private void ExecuteCheckUncheckModel()
+        {
+            this.SelectAllModels = !(this.EngineeringModels.Where(em => !em.IsSelected).Count() > 0);
+        }
+
+        /// <summary>
+        /// Log message to console/output panel
+        /// </summary>
+        /// <param name="message"></param>
+        private void LogMessage(string message)
```

Comment:
not relevant, logging should be somehow injected or use the things that come with the tool.